docs/conf.py

```diff
@@ -221,5 +221,23 @@
 # -- application setup -------------------------------------------------------
 
 
+def setup_to_main(app, pagename, templatename, context, doctree):
+    def to_main(link: str) -> str:
+        """Transform "edit on github" links and make sure they always point to the main branch
+
+        Args:
+            link: the link to the github edit interface
+
+        Returns:
+            the link to the tip of the main branch for the same file
+        """
+        links = link.split("/")
+        idx = links.index("edit")
+        return "/".join(links[: idx + 1]) + "/main/" + "/".join(links[idx + 2 :])
```

Comment:
just a note here that GitHub repos are not guaranteed to have a `main` branch. Also IDK what are the default branch names for GitLab / Bitbucket?  To do this comprehensively we might need to add a config value for default branch name.

Reply:
oh I know that, that's why this extention is living only in our doc and not in the lib itself. It seemed to me that the issue was our doc not all the docs